### DISCUSSION TITLE
Best way to generalize and respecialize with 4.2?

### DISCUSSION THREAD

**Discussion by arkadianriver** [dita-ot/dita-ot#4492] [Q&A] [answered]
Hiya!

I saw that back in 2.0, the team moved generalize functionality into the normalize plugin, but when I use the `dita` transtype, it doesn't generalize. How do I use normalize to generalize.. is there some kind of parameter?

I did found talk of the `generalize.xsl` from older versions, so I pulled it out of 1.5.4 zip and it works well, but I'm wondering if it might be missing something that modern versions might have. I guess as long as I use it with the modern dost jar it should be okay?

What about the round trip; how would I go about respecializing?

Thanks for any tips :O)

**Comment by jelovirt** [comment, marked as answer]
The normalize plug-in was never intended to handle generalization, so the functionality is not missing. I think the original `generalize.xsl` is useable as is, but it's just a matter of how it's used inside or outside DITA-OT.

**Reply by arkadianriver** [reply]
Oh, okay thanks, Jarno. I misunderstood the comments in [this PR from way back](https://github.com/dita-ot/dita-ot/pull/2159).

When I noticed the org.dita-ot.dost.GenUtils namespace in `generalize.xsl`, I thought there was some Java that exposed the class attributes, but as I step through generalize.xsl on a DITA file with the XSL debugger in Oxygen, I discover there's no connection to DITA-OT at all; it seems the DITA class and other attributes are available to the Saxon transform simply by their DTD definitions and the nature of XML.. cool!
